### PR TITLE
PE 810 Add lastSync var

### PR DESCRIPTION
**Pull request by jdaev**
Only sync after context switch if time elapsed since last sync is greater than 5 minutes.

### CODE REVIEW

File: lib/blocs/sync/sync_cubit.dart [hunk 1]
```diff
@@ -46,16 +48,22 @@ class SyncCubit extends Cubit<SyncState> {
...
         .listen((_) {});
     startSync();
   }
 
   void restartSyncOnFocus() {
-    whenBrowserTabIsUnhidden(() => Future.delayed(Duration(seconds: 2))
-        .then((value) => createSyncStream()));
+    whenBrowserTabIsUnhidden(() {
+      if (_lastSync != null &&
+          DateTime.now().difference(_lastSync!).inMinutes <
```
Comment (arielmelendez) [inline lib/blocs/sync/sync_cubit.dart:61]: Surprising that Dart makes you force unwrap _lastSync after having just checked its nullness. Is it definitely necessary?

Comment (jdaev) [reply lib/blocs/sync/sync_cubit.dart:61]: Yeah, it's like that unless I separate it into two if conditions

File: lib/blocs/sync/sync_cubit.dart [hunk 2]
```diff
@@ -46,16 +48,22 @@ class SyncCubit extends Cubit<SyncState> {
...
+              kSyncTimerDuration) {
+        return;
+      }
+      Future.delayed(Duration(seconds: 2)).then((value) => createSyncStream());
```
Comment (arielmelendez) [inline lib/blocs/sync/sync_cubit.dart:65]: I know this was here before but what's the significance of the 2 seconds? Sounds like a hack. And either way it should be documented here.

**Review by arielmelendez — APPROVED** [review]

File: lib/blocs/sync/sync_cubit.dart (hunk 2 above)
Comment (jdaev) [reply lib/blocs/sync/sync_cubit.dart:65]: Just a slight delay so that it doesn't automatically sync after tab switch.

Comment (jdaev) [reply lib/blocs/sync/sync_cubit.dart:65]: It's an arbitrary value

**Pull request merged**